streams/shared/src/main/scala/zio/stream/ZSink.scala

```diff
@@ -1295,6 +1297,78 @@ object ZSink extends ZSinkPlatformSpecific {
         }
     }
 
+  /**
+   * Splits strings on newlines. Handles both `\r\n` and `\n`.
+   */
+  final val splitLines: ZSink[Any, Nothing, String, String, Chunk[String]] =
+    new SinkPure[Nothing, String, String, Chunk[String]] {
+      type State = (Chunk[String], Option[String], Boolean)
```

Comment:
On an urelated note, this is probably what the stepless sinks' code will look like after the refactoring. :)